pkg/didcomm/common/service/service_test.go

```diff
@@ -0,0 +1,57 @@
+package service
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDIDCommMsg_ThreadID(t *testing.T) {
+	tests := []struct {
+		msg DIDCommMsg
+		val string
+		err string
+	}{{
+		msg: DIDCommMsg{},
+		err: "cannot unmarshal @id and ~thread: error=unexpected end of JSON input",
+	}, {
+		msg: DIDCommMsg{Payload: []byte(`{"@id":"ID"}`)},
+		val: "ID",
+		err: "",
+	}, {
+		msg: DIDCommMsg{Payload: []byte(`{"@id":"ID","~thread":{"thid":"thID"}}`)},
+		val: "thID",
+		err: "",
+	}, {
+		msg: DIDCommMsg{Payload: []byte(`{"~thread":{"thid":"thID"}}`)},
+		val: "thID",
+		err: "",
+	}, {
+		msg: DIDCommMsg{Payload: []byte(`{}`)},
+		val: "",
+		err: "",
```

Comment:
how come an empty JSON doesn't return an error while an empty array does (TC flow)

Reply:
empty JSON is valid json, but values are empty

Reply:
got it ) `{}` vs `[]` 
`{}` - can be structure, map or interface
 `[]` - only array or interface 
we can not unmarshal list into the structure